### ISSUE TITLE
Scientific Metadata with / without '.value' (or ' .v')

### ISSUE THREAD

**Issue by linupi** [SciCatProject/scicat-backend-next#138]
### Summary

when working on https://github.com/SciCatProject/scicat-backend-next/pull/119  we saw that for the new SciCat backend there is currently no support for metadata outside the '.value', '.unit' scheme. In https://github.com/SciCatProject/scicat-backend-next/issues/119#issuecomment-1201236240_ there is an example of scientific metadata from PSI that falls in this category:

![image](https://user-images.githubusercontent.com/8998708/182163269-32f1df25-672d-4d1b-9047-3ece22762c2a.png)

_Originally posted by @stephan271 in https://github.com/SciCatProject/scicat-backend-next/issues/119#issuecomment-1201236240_

### Expected Behaviour

like in the old backend the frontend search should work also on any string and integer values in the metadata without having a 'unit'  / 'value'.  Further in the example above 'v' and 'u' instead of 'value' and 'unit'.

I have the feeling it would make sense to provide a sort of generic helper to deal with the different options of having .value, .value+.unit or none of them. Currently '.value' is hardcoded in the new backend code e.g. 

<pre>src/common/utils.ts<font color="#06989A">:</font>20<font color="#06989A">:</font>    return { valueSI: Number(quantity<font color="#CC0000"><b>.value</b></font>), unitSI: quantity.unit };
src/common/utils.ts<font color="#06989A">:</font>92<font color="#06989A">:</font>    const matchKeyMeasurement = `scientificMetadata.${lhs}<font color="#CC0000"><b>.value</b></font>SI`;
src/common/utils.ts<font color="#06989A">:</font>97<font color="#06989A">:</font>        scientificFilterQuery[`${matchKeyGeneric}<font color="#CC0000"><b>.value</b></font>`] = { $eq: rhs };
src/common/utils.ts<font color="#06989A">:</font>102<font color="#06989A">:</font>        scientificFilterQuery[`${matchKeyGeneric}<font color="#CC0000"><b>.value</b></font>`] = { $eq: escapedRhs };
src/common/utils.ts<font color="#06989A">:</font>111<font color="#06989A">:</font>          scientificFilterQuery[`${matchKeyGeneric}<font color="#CC0000"><b>.value</b></font>`] = { $eq: rhs };
src/common/utils.ts<font color="#06989A">:</font>121<font color="#06989A">:</font>          scientificFilterQuery[`${matchKeyGeneric}<font color="#CC0000"><b>.value</b></font>`] = { $gt: rhs };
src/common/utils.ts<font color="#06989A">:</font>131<font color="#06989A">:</font>          scientificFilterQuery[`${matchKeyGeneric}<font color="#CC0000"><b>.value</b></font>`] = { $lt: rhs };
src/datasets/interceptors/fullquery.interceptor.ts<font color="#06989A">:</font>34<font color="#06989A">:</font>                `${lhs}<font color="#CC0000"><b>.value</b></font>`,
src/datasets/interceptors/fullquery.interceptor.ts<font color="#06989A">:</font>46<font color="#06989A">:</font>                  `${lhs}<font color="#CC0000"><b>.value</b></font>`,
</pre>

**Comment by nitrosx** [comment]
@linupi : I agree with your suggestion about helper functions when retrieving, updating and query metadata.
Personally I would allow the following 2 cases:
- unitless metadata with the following syntax:
```
property : value
```
or
```
property : { 'value': value, 'unit': '' }
```
- metadata with unit
```
property : { 'value' : value, 'unit': unit }
```

**Comment by bpedersen2** [comment]
It looks like 'string' metadata use the second form ( value + empty unit) when creating them from the  frontend.

**Comment by bpedersen2** [comment]
ping...

**Comment by rkweehinzmann** [comment]
Going through this issue in more detail it seems to me we do not need any generalised helper to deal with different cases as suggested by @linupi back in 2022, but rather implement value + optional unit as mentioned here by @nitrosx : property : { 'value': value, 'unit': '' } This should solve #119, ie. the issue of finding meta data by inputting only part of a string without any unit, right?

**Comment by linupi** [comment]
well back in 2022 I just described the discrepancy of what existed in the _old_ backend compared to the new one. The old one generally worked with `property: 'value'` as well as `property:{value:'value', unit:''}` for the exact matching of strings so it was just natural to apply the same logic also for the partial matching.
In the end it is a design choice if `property: 'value'` would be acceptable or not - and i could probably survive with both options. To me it still makes sense to have such an option since `property:{value:'my string', unit:'an_existing_unit'}` would need to be carefully filtered and forbidden I guess. Further, if you look at what @stephan271 posted in https://github.com/SciCatProject/scicat-backend-next/issues/119#issuecomment-1201236240_ 
an enforcement of  `property:{value:'value', unit:''}` would break with the current practice 

![image](https://user-images.githubusercontent.com/8998708/182163269-32f1df25-672d-4d1b-9047-3ece22762c2a.png)

so in conclusion I strongly tend to suggest to not enforce `unit`  `value` pairs, as this is not always a correct representation. Looking at the ROI in the last rows of above's screenshot one could even debate if units should even be enforced on numeric values as well.

**Comment by nitrosx** [comment]
I think we should refactor the code to accept with and without the value field to make it backward compatible with alder entries.
At ESS, we are going to use the value and unit fields.

**Comment by nitrosx** [comment]
I propose that SciCat can correctly work and represent the following cases:
- strings
`property : value`
`property: { 'value': 'value', 'unit': '' }`
`property: { 'v': 'value', 'u': '' }`
- numbers
`property : value`
`property: { 'value': 'value', 'unit': '' }`
`property: { 'v': 'value', 'u': '' }`
- quantities
`property: { 'value': 'value', 'unit': 'unit' }`
`property: { 'v': 'value', 'u': 'unit' }`
- nested objects
```
property: { 
  'sub-property-1' : any of the option above
  'sub-property-2' : any of the option above
  ...
}
```
```
property: { 
  'value' : {
    'sub-property-1' : any of the option above
    'sub-property-2' : any of the option above
    ...
  }
  'unit': ''
}
```

I will leave it up to each facilities if they want to enforce any of the above formats and which ones
